test/unit/ipam_test.rb

```diff
@@ -45,6 +45,49 @@ class IPAMTest < ActiveSupport::TestCase
     end
   end
 
+  context 'random db' do
+    test "should find unused IPv4" do
+      subnet = FactoryGirl.create(
+        :subnet_ipv4, :name => 'my_subnet',
+        :network => '10.0.0.0',
+        :mask => '255.0.0.0',
+        :ipam => IPAM::MODES[:random_db])
+      ipam = IPAM::RandomDb.new(:subnet => subnet)
+      assert_match /^10\./, ipam.suggest_ip
+    end
+
+    test "should return IPv4 based on MAC if provided" do
+      subnet = FactoryGirl.create(
+        :subnet_ipv4, :name => 'my_subnet',
+        :network => '10.0.0.0',
+        :mask => '255.0.0.0',
+        :ipam => IPAM::MODES[:random_db])
+      ipam1 = IPAM::RandomDb.new(:subnet => subnet, :mac => "AA:BB:CC:DD:EE:FF")
+      ipam2 = IPAM::RandomDb.new(:subnet => subnet, :mac => "AA:BB:CC:DD:EE:FF")
+      assert_equal ipam1.suggest_ip, ipam2.suggest_ip
+    end
+
+    test "should find unused IPv6" do
+      subnet = FactoryGirl.create(
+        :subnet_ipv6, :name => 'my_subnet',
+        :network => '2001:db8::',
+        :ipam => IPAM::MODES[:db])
+      ipam = IPAM::RandomDb.new(:subnet => subnet)
+      assert_match /^2001:db8::/, ipam.suggest_ip
+    end
+
+    test "should find the only possible IPv4" do
+      subnet = FactoryGirl.create(
+        :subnet_ipv4, :name => 'my_subnet',
+        :network => '192.168.11.0',
+        :from => '192.168.11.5',
+        :to => '192.168.11.5',
+        :ipam => IPAM::MODES[:random_db])
+      ipam = IPAM::RandomDb.new(:subnet => subnet, :excluded_ips => (2..254).map{|x| "192.168.2.#{x}"})
```

Comment:
Another leftover, anyway it is a good idea to test excluded IPs. Will do that as well as test for hard limit of iterations done.